wallet/src/ui/app/redux/slices/transaction-bytes-requests/index.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSlice,
+} from '@reduxjs/toolkit';
+
+import type { TransactionResponse } from '@mysten/sui.js';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import type { TransactionBytesRequest } from '_payloads/transactions';
+import type { RootState } from '_redux/RootReducer';
+import type { AppThunkConfig } from '_store/thunk-extras';
+
+const txBytesRequestsAdapter = createEntityAdapter<TransactionBytesRequest>({
+    sortComparer: (a, b) => {
+        const aDate = new Date(a.createdDate);
+        const bDate = new Date(b.createdDate);
+        return aDate.getTime() - bDate.getTime();
+    },
+});
+
+export const respondToTransactionRequest = createAsyncThunk<
+    {
+        txRequestID: string;
+        approved: boolean;
+        txResponse: TransactionResponse | null;
+    },
+    { txRequestID: string; approved: boolean },
+    AppThunkConfig
+>(
+    'respond-to-transaction-bytes-request',
+    async (
+        { txRequestID, approved },
+        { extra: { background, api, keypairVault }, getState }
+    ) => {
+        const state = getState();
+        const txBytesRequest = txBytesRequestsSelectors.selectById(state, txRequestID);
+        if (!txBytesRequest) {
+            throw new Error(`TransactionBytesRequest ${txRequestID} not found`);
+        }
+        let txResult: TransactionResponse | undefined = undefined;
+        let tsResultError: string | undefined;
+        if (approved) {
+            const signer = api.getSignerInstance(keypairVault.getKeyPair());
+            try {
+                txResult = await signer.signAndExecuteTransaction(txBytesRequest.txBytes);
+            } catch (e) {
+                tsResultError = (e as Error).message;
+            }
+        }
+        background.sendTransactionRequestResponse(
+            txRequestID,
+            approved,
+            txResult,
+            tsResultError
+        );
+        return { txRequestID, approved: approved, txResponse: null };
+    }
+);
+
+const slice = createSlice({
+    name: 'transaction-bytes-requests',
+    initialState: txBytesRequestsAdapter.getInitialState({ initialized: false }),
+    reducers: {
+        setTransactionBytesRequests: (
+            state,
+            { payload }: PayloadAction<TransactionBytesRequest[]>
+        ) => {
+            txBytesRequestsAdapter.setAll(state, payload);
```

Comment:
here I saw an error of 
```
No overload matches this call.
  Overload 1 of 2, '(state: EntityState<TransactionBytesRequest>, entities: readonly TransactionBytesRequest[] | Record<EntityId, TransactionBytesRequest>): EntityState<...>', gave the following error.
    Argument of type 'WritableDraft<EntityState<TransactionBytesRequest> & { initialized: boolean; }>' is not assignable to parameter of type 'EntityState<TransactionBytesRequest>'.
Types of property 'entities' are incompatible.
Type 'WritableDraft<Dictionary<TransactionBytesRequest>>' is not assignable to type 'Dictionary<TransactionBytesRequest>'.
          'string' index signatures are incompatible.
Type 'WritableDraft<TransactionBytesRequest> | undefined' is not assignable to type 'TransactionBytesRequest | undefined'.
Type 'WritableDraft<TransactionBytesRequest>' is not assignable to type 'TransactionBytesRequest'.
Types of property 'txBytes' are incompatible.
                  Property 'data' is missing in type 'WritableDraft<Base64DataBuffer>' but required in type 'Base64DataBuffer'.
Overload 2 of 2, '(state: EntityState<TransactionBytesRequest>, entities: { payload: readonly TransactionBytesRequest[] | Record<EntityId, TransactionBytesRequest>; type: string; }): EntityState<...>', gave the following error.
Argument of type 'WritableDraft<EntityState<TransactionBytesRequest> & { initialized: boolean; }>' is not assignable to parameter of type 'EntityState<TransactionBytesRequest>'.
```

Reply:
contrary to this, when I hover over transaction-requests/index.ts, I saw same thing while no errors there.